### PR TITLE
Geoffl/error handling

### PR DESCRIPTION
**Pull request by geofflorne**
add some more robust error handling and remove and old function.

### CODE REVIEW

File: router.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package main
 import (
 	"encoding/json"
 	"fmt"
-	"log"
+	//"log"
```
Comment (JakeCooper) [inline router.go]: Commented import :(

Comment (geofflorne) [reply router.go]: Oops I'll fix that

**Comment by geofflorne** [comment]
Fixed @JakeCooper

File: router.go [hunk 2]
```diff
@@ -168,12 +180,13 @@ func regHandler(w http.ResponseWriter, r *http.Request) {
...
+		if err != nil{
+			failWithStatusCode(err, "Problem with update query", w, http.StatusInternalServerError)
+			return
+		}
 		numRows, err := res.RowsAffected()
 		if numRows < 1 {
-			failOnError(err, "Unable to update new user")
-			w.WriteHeader(http.StatusConflict)
-			fmt.Fprintf(w, "Server Error")
+			failWithStatusCode(err, "Unable to update user", w, http.StatusInternalServerError)
```
Comment (tzinck) [inline router.go]: numRows could be less than one, with with err being nil (ie, the update query runs successfully but does not affect any rows, and rowsAffected is called successfully and returns 0), but this won't send any response in that case.

**Review by JakeCooper — APPROVED** [review]

**Pull request merged**